pkg/api/validation/dynakube/oneagent.go

```diff
@@ -191,3 +197,53 @@ func isOneAgentVersionValid(_ context.Context, _ *Validator, dk *dynakube.DynaKu
 
 	return ""
 }
+
+func hasNoDuplicatOneAgentArguments(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	args := dk.OneAgentArguments()
+	if args == nil {
+		return ""
+	}
+
+	for key, values := range args {
+		if key != "--set-host-property" && key != "--set-host-tag" && len(values) > 1 {
+			return fmt.Sprintf(errorDuplicateOneAgentArgument, key)
+		} else if key == "--set-host-tag" {
+			if duplicatedTags := findDuplicates(values); len(duplicatedTags) > 0 {
+				return fmt.Sprintf(errorSameHostTagMultipleTimes, duplicatedTags)
+			}
+		}
+	}
+
+	return ""
+}
+
+func hasNoHostIdSourceArgument(_ context.Context, _ *Validator, dk *dynakube.DynaKube) string {
+	args := dk.OneAgentArguments()
+	if args == nil {
+		return ""
+	}
+
+	for key := range args {
+		if dk.CloudNativeFullstackMode() && key == "--set-host-id-source" {
+			return errorHostIdSourceArgumentInCloudNative
+		}
+	}
+
+	return ""
+}
+
+func findDuplicates[S ~[]E, E comparable](s S) []E {
+	seen := make(map[E]bool)
+
+	duplicates := make([]E, 0)
+
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			seen[val] = true
+		} else {
+			duplicates = append(duplicates, val)
+		}
+	}
+
+	return duplicates
+}
```

Comment:
you are collecting the duplicates in a way that you duplicating them in the output 😅
- which is IMO not that helpful

so if you have:
- `[]string{"foo", "foo", "foo", "foo"}` the output will be `[]string{"foo", "foo", "foo"}` -> OK? kinda confusing
- `[]string{"foo", "foo", "bar", "bar"}` the output will be `[]string{"foo", "bar"}` -> IMO this is what you want (or this is the least confusing output)
- `[]string{"foo", "foo", "bar", "bar", "foo"}` the output will be `[]string{"foo", "bar", "foo"}` -> well, this is confusing 😅

you could just do:
```suggestion
func findDuplicates[S ~[]E, E comparable](s S) []E {
	seen := make(map[E]bool)

	duplicates := make([]E, 0)

	for _, val := range s {
		if _, ok := seen[val]; !ok {
			seen[val] = true
            duplicates = append(duplicates, val)
		}
	}

	return duplicates
}
```

but IMO, this is still a bit of overengineering, I think it would be enough to tell the user that `you defined duplicate --set-host-tag args` the rest they can figure out easily

Reply:
I agree with @0sewa0. It's overengeniring for mostly nothing.

Reply:
Telling the user as exactly as possible what the problem is improves UX, so this clearly has value to the customer.
But the duplication of tags in the error message needs some fixing, ofc.

Reply:
Fixed in [5d093cf](https://github.com/Dynatrace/dynatrace-operator/pull/4377/commits/5d093cf3fe7092cda67d5f0f73208e02a81e2716)